### PR TITLE
[UTOPIA-527] Introducing unit testing with pia-intake

### PR DESCRIPTION
**Pull request by kushal-arora-fw**
# Description

This PR includes the following proposed change(s):

- Unit testing base structure [moving to `test` directory for future tests]
- pia-intake unit tests

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

## How Has This Been Tested?

run `npm run test` inside `src/backend`

## Development Dependency Working Agreement
- [x] My code DOES NOT include the importing of new dependencies into the DPIA ecosystem
- [ ] My code DOES include the importing of new dependencies into the DPIA ecosystem
**If new dependencies are being introduced to the DPIA ecosystem:**
- [ ] The functionality of the dependency drastically reduces code complexity and makes my changes more easily maintainable and readible 
- [ ] The dependency being introduced does not contain multiple layers of nested dependencies introducing maintainability complexity to the DPIA ecosystem

## Frontend Development Changes
- [x] N/A
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to project documentation or diagrams that reflect my changes
- [ ] New and existing unit tests pass locally with my changes
- [ ] My code follows Airbnb React Style Guidelines

## API Development Changes
- [ ] N/A
- [x] I have performed a self-review of my own code
- [x] My code follows standards and practices outlined in the BC Government API Development Guidelines
- [x] New and existing unit tests pass locally with my changes
- [ ] My changes includes Swagger documentation updates that reflect the changes I am introducing

## Definition of Done

Definition of Done

### CODE REVIEW

File: src/backend/test/unit/pia-intake/pia-intake.controller.spec.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  keycloakUserMock,
+} from 'test/util/mocks/data/auth.mock';
+import { createPiaIntakeMock } from 'test/util/mocks/data/pia-intake.mock';
+import { repositoryMock } from 'test/util/mocks/repository/repository.mock';
+import { KeycloakUser } from 'src/modules/auth/keycloak-user.model';
+import { PiaIntakeService } from 'src/modules/pia-intake/pia-intake.service';
+import { piaIntakeServiceMock } from 'test/util/mocks/services/pia-intake.service.mock';
+import { UnauthorizedException } from '@nestjs/common';
+
+describe('PiaIntakeController', () => {
```
Comment (antsand) [inline src/backend/test/unit/pia-intake/pia-intake.controller.spec.ts:21]: let us add proper comments before a test case.

**Review by antsand — COMMENTED** [review]
For test cases can we have comment placed before every test case. New developers generally don't run test cases or modify existing test cases due to lack of description.

A sample description template
/*
[Description] - To check the pia -intake route 
[input] - url or function call
[Expected output] -  Route to the right API view
[ What does it check] - Routes, API calls, API sanity, login info 
*/
 
We can discuss more on this.

**Comment by kushal-arora-fw** [comment]
@antsand  - updated test description. Feel free to give it a look

**Review by antsand — APPROVED** [review]
When a new user comes to this test file, what are they supposed to do? How do they run it. It would be super helpful to spoon feed a little. 

Having it on  a wiki almost guarantee this test file won't be executed (in the long term).

**Pull request merged**